### PR TITLE
Add default values to `proxy-*-connect-timeout` annotations docs

### LINKED ISSUE #12154: Add default connection timeout values to proxy configuration documentation.
### What is the issue?

It's pretty hard to figure out what the default values are for some of the proxy configurations. Most (but not all) of the fields in the proxy configuration doc have an ending sentence declaring the default value.

Add that for the inbound and outbound connect timeouts in the proxy documentation too.

### How can it be reproduced?

N/A

### Logs, error output, etc

N/A

### output of `linkerd check -o short`

N/A

### Environment

N/A

### Possible solution

_No response_

### Additional context

_No response_

### Would you like to work on fixing this bug?

yes

### PR DESCRIPTION
**Pull request by doubletooth**
Finding these values in the docs site is pretty painful

Update the documentation to follow other proxy fields in providing a default value at the end of the description for the field

Fixes #12154

### CODE REVIEW

**Review by alpeb — COMMENTED** [review]
This is useful, thanks!
I think however that the default for inbound connect timeout is 100ms (see charts/linkerd-control-plane/values.yaml. Also can you use the same unit (ms) for the outbound connect timeout?

File: cli/cmd/doc.go [hunk 1]
```diff
@@ -238,11 +238,11 @@ func generateAnnotationsDocs() []annotationDoc {
 		},
 		{
 			Name:        k8s.ProxyOutboundConnectTimeout,
-			Description: "Used to configure the outbound TCP connection timeout in the proxy",
+			Description: "Used to configure the outbound TCP connection timeout in the proxy. Defaults to `1s`",
```
Comment (alpeb) [inline cli/cmd/doc.go:241]:
```suggestion
			Description: "Used to configure the outbound TCP connection timeout in the proxy. Defaults to `1000ms`",
```

File: cli/cmd/doc.go [hunk 2]
```diff
@@ -238,11 +238,11 @@ func generateAnnotationsDocs() []annotationDoc {
...
 		},
 		{
 			Name:        k8s.ProxyInboundConnectTimeout,
-			Description: "Inbound TCP connection timeout in the proxy",
+			Description: "Inbound TCP connection timeout in the proxy. Defaults to `300ms`",
```
Comment (alpeb) [inline cli/cmd/doc.go:245]:
```suggestion
			Description: "Inbound TCP connection timeout in the proxy. Defaults to `100ms`",
```

**Review by alpeb — APPROVED** [review]

**Comment by doubletooth** [comment]
Hey @alpeb thanks for applying the suggestion and merging in the submission! I really appreciate it. Sorry I couldn't respond sooner.

**Pull request merged**